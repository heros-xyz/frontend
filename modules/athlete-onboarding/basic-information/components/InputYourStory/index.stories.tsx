import type { Meta, StoryObj } from "@storybook/react";
import { Formik } from "formik";
import InputYourStory from ".";

const meta: Meta<typeof InputYourStory> = {
  title: "Athlete/PageInfo/YourStory",
  component: InputYourStory,
  tags: ["autodocs"],
  parameters: {
    layout: "fullscreen",
  },
};

export default meta;
type Story = StoryObj<typeof InputYourStory>;

export const InputYourStoryStory: Story = {
  render: () => (
    <Formik onSubmit={console.log} initialValues={{ nationality: "" }}>
      <InputYourStory onSubmit={console.log} />
    </Formik>
  ),
};
