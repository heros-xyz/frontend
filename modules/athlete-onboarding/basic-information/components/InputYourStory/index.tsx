import { Box, Input, Text } from "@chakra-ui/react";
import { useState } from "react";
import { useFormikContext } from "formik";
import { ArrowRight } from "@/components/svg/ArrowRight";
import FanOnboardingWrapper from "@/components/ui/HerosOnboardingWrapper";
import { YourStoryIcon } from "@/components/svg/YourStoryIcon";
import ErrorMessage from "@/components/common/ErrorMessage";
import { IValuesTypes } from "../../hooks";

interface InputYourStoryProps {
  onSubmit?: () => void;
}

const InputYourStory: React.FC<InputYourStoryProps> = ({ onSubmit }) => {
  const [submitted, setSubmitted] = useState(false);
  const { values, errors, handleSubmit, handleChange } =
    useFormikContext<IValuesTypes>();

  const submitForm = () => {
    setSubmitted(true);
    handleSubmit();

    if (!errors.story && values.story) {
      onSubmit && onSubmit();
    }
  };

  return (
    <FanOnboardingWrapper
      Icon={<YourStoryIcon w="full" h="full" />}
      textButton="SUBMIT"
      onSubmit={submitForm}
      title="BASIC INFORMATION"
    >
      <Box mb={{ base: 4, lg: 10 }} color="black.ish">
        <Box fontSize={{ lg: "xl" }} fontWeight="500" mb={1.5}>
          Tell Your Story (500 characters max)
          <Text as="span" color="error.dark">
            {" "}
            *
          </Text>
        </Box>
        <Text as="p" fontSize="xs" mb={2}>
          Tell a compelling story of yourself to inspire potential fans.
        </Text>
        <Input
          variant="flushed"
          placeholder="Tell Your Story"
          borderColor="primary"
          name="story"
          fontSize={{ base: "sm" }}
          isInvalid={Boolean(errors.story && submitted)}
          value={values?.story}
          onChange={handleChange}
        />
        <ErrorMessage
          condition={errors.story && submitted}
          errorMessage={errors.story}
        />
      </Box>
    </FanOnboardingWrapper>
  );
};

export default InputYourStory;
