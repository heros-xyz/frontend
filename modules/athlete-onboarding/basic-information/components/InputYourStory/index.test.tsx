import { render } from "@testing-library/react";
import { Formik } from "formik";
import InputYourStory from ".";

it("renders Input Your Story unchanged", () => {
  const { container } = render(
    <Formik onSubmit={console.log} initialValues={{ story: "" }}>
      <InputYourStory onSubmit={console.log} />
    </Formik>
  );
  expect(container).toMatchSnapshot();
});
