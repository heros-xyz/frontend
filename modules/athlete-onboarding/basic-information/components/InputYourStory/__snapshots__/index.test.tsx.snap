// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Input Your Story unchanged 1`] = `
<div>
  <div
    class="css-16tif5l"
  >
    <div
      class="chakra-container css-10klw3m"
    >
      <div
        class="css-12ij4x2"
      >
        BASIC INFORMATION
      </div>
      <div
        class="css-8qr4sb"
      >
        <div
          class="css-tayfq6"
        >
          <div
            class="css-6ywt87"
          >
            <svg
              class="chakra-icon css-h2mabo"
              focusable="false"
              viewBox="0 0 151 150"
              xmlns="http://www.w3.org/2000/svg"
            >
              <circle
                cx="75.5"
                cy="75"
                fill="white"
                r="74.5"
                stroke="#DDDDDD"
              />
              <g
                clip-path="url(#clip0_1504_19506)"
              >
                <path
                  d="M76.1993 104.4C76.0993 104.3 75.5993 103.5 75.5993 103.4C75.1993 102.8 74.8993 101.1 74.4993 100.5C74.0993 99.8998 73.5993 99.2998 73.1993 98.5998C72.3993 97.2998 71.5993 95.9998 71.0993 94.6998C70.5993 93.3998 69.9993 92.0998 69.3993 90.8998C69.2993 90.8998 69.0993 90.8998 68.9993 90.6998C68.1993 89.8998 67.6993 88.8998 67.2993 87.7998C66.2993 89.1998 65.5993 90.1998 65.2993 90.6998C63.5993 93.4998 64.7993 97.2998 65.1993 99.7998C65.6993 102.3 69.5993 100.7 69.5993 100.7C66.7993 108.1 72.9993 106.1 72.9993 106.1C72.6993 112.9 77.7993 111 77.7993 111C78.3993 110.7 79.5993 109.1 80.1993 108.8C80.1993 108.7 78.4993 107.1 77.6993 105.7"
                  fill="#FEBF2E"
                />
                <path
                  d="M76.1993 104.4C76.0993 104.3 75.5993 103.5 75.5993 103.4C75.1993 102.8 74.8993 101.1 74.4993 100.5C74.0993 99.8998 73.5993 99.2998 73.1993 98.5998C72.3993 97.2998 71.5993 95.9998 71.0993 94.6998C70.5993 93.3998 69.9993 92.0998 69.3993 90.8998C69.2993 90.8998 69.0993 90.8998 68.9993 90.6998C68.1993 89.8998 67.6993 88.8998 67.2993 87.7998C66.2993 89.1998 65.5993 90.1998 65.2993 90.6998C63.5993 93.4998 64.7993 97.2998 65.1993 99.7998C65.6993 102.3 69.5993 100.7 69.5993 100.7C66.7993 108.1 72.9993 106.1 72.9993 106.1C72.6993 112.9 77.7993 111 77.7993 111C78.3993 110.7 79.5993 109.1 80.1993 108.8C80.1993 108.7 78.4993 107.1 77.6993 105.7"
                  stroke="black"
                  stroke-miterlimit="10"
                  stroke-width="1.5"
                />
                <path
                  d="M92.8004 89.9998C92.7004 85.1998 89.4004 85.4998 89.4004 85.4998C90.8004 80.0998 86.0004 78.3998 86.0004 78.3998C87.0004 72.7998 81.6004 68.2998 81.6004 68.2998C81.6004 68.2998 76.1004 76.9998 71.4004 83.6998C72.3004 83.4998 73.8004 85.6998 74.3004 86.4998C75.7004 88.3998 75.4004 89.3998 76.4004 91.4998C77.4004 93.5998 80.8004 99.7998 81.4004 101.3C81.4004 101.3 82.4004 106.5 82.9004 107.9C89.2004 102.5 93.0004 94.5998 92.8004 89.9998Z"
                  fill="#FAC484"
                />
                <path
                  d="M92.8004 89.9998C92.7004 85.1998 89.4004 85.4998 89.4004 85.4998C90.8004 80.0998 86.0004 78.3998 86.0004 78.3998C87.0004 72.7998 81.6004 68.2998 81.6004 68.2998C81.6004 68.2998 76.1004 76.9998 71.4004 83.6998C72.3004 83.4998 73.8004 85.6998 74.3004 86.4998C75.7004 88.3998 75.4004 89.3998 76.4004 91.4998C77.4004 93.5998 80.8004 99.7998 81.4004 101.3C81.4004 101.3 82.4004 106.5 82.9004 107.9C89.2004 102.5 93.0004 94.5998 92.8004 89.9998Z"
                  stroke="black"
                  stroke-miterlimit="10"
                  stroke-width="1.5"
                />
                <path
                  d="M78.6992 62.4001C75.1992 59.7001 66.8992 60.2001 61.6992 60.8001C62.1992 62.0001 62.5992 63.2001 62.9992 64.3001L69.1992 67.3001C69.1992 67.3001 68.5992 70.5001 68.0992 73.5001C68.5992 73.7001 68.9992 74.1001 69.2992 74.7001C70.2992 76.7001 70.6992 78.8001 71.2992 80.9001C71.3992 81.2001 71.3992 81.6001 71.2992 81.9001C73.7992 83.2001 77.1992 83.1001 79.3992 78.6001C83.2992 70.4001 83.8992 66.4001 78.6992 62.4001Z"
                  fill="#FAC484"
                />
                <path
                  d="M75.6992 84.9001L73.9992 62.1001L61.6992 61.6001C62.0992 62.9001 65.8992 71.7001 67.4992 74.0001C69.1992 76.4001 70.0992 77.7001 70.7992 80.5001C71.1992 82.1001 72.3992 83.1001 72.7992 84.8001C73.2992 86.0001 73.5992 85.4001 74.2992 86.5001L75.6992 84.9001Z"
                  fill="#FAC484"
                />
                <path
                  d="M71.6997 94.7999C71.2997 93.7999 70.9997 92.6999 70.5997 91.6999C69.6997 91.4999 67.7997 85.9999 67.3997 84.1999C66.1997 84.3999 65.5997 83.2999 65.2997 81.7999C64.3997 78.1999 56.9997 65.3999 55.5997 61.8999C51.6997 63.1999 32.0997 70.0999 30.4997 74.3999C28.4997 79.6999 35.4997 99.8999 38.7997 112.6C40.1997 117.7 76.2997 111.1 76.2997 111.1L76.8997 104.3C75.8997 103.1 74.3997 99.7999 73.6997 98.3999C72.9997 97.1999 72.0997 96.0999 71.6997 94.7999Z"
                  fill="#FAC484"
                />
                <path
                  d="M65.599 92.5C64.299 93.7 64.299 96.6 64.899 98.2C65.799 100.3 67.899 100.6 69.899 100C69.299 101.2 68.699 103.5 69.499 104.8C70.299 106 72.099 106.2 73.199 105.6C73.099 107.3 72.899 108.9 74.499 110.1C76.199 111.3 78.299 110.7 79.599 109.2"
                  fill="#FAC484"
                />
                <path
                  d="M65.599 92.5C64.299 93.7 64.299 96.6 64.899 98.2C65.799 100.3 67.899 100.6 69.899 100C69.299 101.2 68.699 103.5 69.499 104.8C70.299 106 72.099 106.2 73.199 105.6C73.099 107.3 72.899 108.9 74.499 110.1C76.199 111.3 78.299 110.7 79.599 109.2"
                  stroke="black"
                  stroke-miterlimit="10"
                  stroke-width="1.5"
                />
                <path
                  d="M53.6988 43.5999L50.7988 37.3999C50.1988 36.0999 48.4988 35.3999 47.0988 35.9999L46.4988 36.1999C44.9988 36.7999 44.1988 38.4999 44.8988 39.9999L47.7988 46.1999L53.6988 43.5999Z"
                  fill="#F2696A"
                />
                <path
                  d="M53.6988 43.5999L50.7988 37.3999C50.1988 36.0999 48.4988 35.3999 47.0988 35.9999L46.4988 36.1999C44.9988 36.7999 44.1988 38.4999 44.8988 39.9999L47.7988 46.1999L53.6988 43.5999Z"
                  stroke="black"
                  stroke-miterlimit="10"
                  stroke-width="1.5"
                />
                <path
                  d="M53.9 44.5L48 47.2L50.1 51.4L56 48.7L53.9 44.5Z"
                  fill="#A4C3DD"
                />
                <path
                  d="M53.9 44.5L48 47.2L50.1 51.4L56 48.7L53.9 44.5Z"
                  stroke="black"
                  stroke-miterlimit="10"
                  stroke-width="1.5"
                />
                <path
                  d="M77.3996 100.3C77.4996 100.1 77.6996 99.9997 77.9996 99.9997L81.2996 100.3L70.7996 79.3997L56.4996 49.6997L50.5996 52.3997L75.5996 103.5L77.3996 100.3Z"
                  fill="#F05253"
                />
                <path
                  d="M77.3996 100.3C77.4996 100.1 77.6996 99.9997 77.9996 99.9997L81.2996 100.3L70.7996 79.3997L56.4996 49.6997L50.5996 52.3997L75.5996 103.5L77.3996 100.3Z"
                  stroke="black"
                  stroke-miterlimit="10"
                  stroke-width="1.5"
                />
                <path
                  d="M70.9004 61.3999C78.5004 62.1999 80.8004 63.5999 81.7004 65.2999C83.1004 67.8999 82.8004 70.5999 82.7004 73.4999C82.5004 76.1999 82.1004 79.0999 80.8004 81.4999C79.4004 83.9999 77.0004 88.2999 73.5004 86.2999C69.9004 84.2999 71.4004 76.2999 72.1004 73.3999"
                  fill="#FAC484"
                />
                <path
                  d="M70.9004 61.3999C78.5004 62.1999 80.8004 63.5999 81.7004 65.2999C83.1004 67.8999 82.8004 70.5999 82.7004 73.4999C82.5004 76.1999 82.1004 79.0999 80.8004 81.4999C79.4004 83.9999 77.0004 88.2999 73.5004 86.2999C69.9004 84.2999 71.4004 76.2999 72.1004 73.3999"
                  stroke="black"
                  stroke-miterlimit="10"
                  stroke-width="1.5"
                />
                <path
                  d="M83.1992 111.9C84.8992 110.3 86.4992 108.7 88.3992 107.3C88.5992 108.2 88.9992 109.3 89.5992 110.1C90.3992 111.2 92.0992 111.5 93.3992 111.2C94.4992 111 95.8992 109.6 96.9992 109.7C98.1992 109.8 98.9992 111.5 100.299 111.8C101.899 112.2 102.899 111.2 104.299 110.8C104.899 110.6 105.699 110.4 106.299 110.7C106.799 110.9 106.999 111.4 107.599 111.6C108.899 112.1 110.699 111.9 111.999 111.9C114.799 111.8 117.599 111.4 120.399 111.2"
                  stroke="black"
                  stroke-miterlimit="10"
                  stroke-width="1.5"
                />
                <path
                  d="M52.2988 54.6001L75.9988 103"
                  stroke="black"
                  stroke-miterlimit="10"
                  stroke-width="1.5"
                />
                <path
                  d="M80.6996 100.5C80.2996 100.4 77.8996 100.2 77.8996 100.2C77.8996 100.2 76.1996 102.7 76.0996 103.2C77.3996 105.1 82.0996 112.6 83.1996 111.9C84.1996 111.3 81.3996 102.7 80.6996 100.5Z"
                  fill="#F05253"
                />
                <path
                  d="M80.6996 100.5C80.2996 100.4 77.8996 100.2 77.8996 100.2C77.8996 100.2 76.1996 102.7 76.0996 103.2C77.3996 105.1 82.0996 112.6 83.1996 111.9C84.1996 111.3 81.3996 102.7 80.6996 100.5Z"
                  stroke="black"
                  stroke-miterlimit="10"
                  stroke-width="1.5"
                />
                <path
                  d="M80.0996 109.9L83.2996 107.6C83.2996 107.6 84.4996 111.7 83.4996 112.3C82.2996 113.1 80.0996 109.9 80.0996 109.9Z"
                  fill="black"
                />
                <path
                  d="M80.0996 109.9L83.2996 107.6C83.2996 107.6 84.4996 111.7 83.4996 112.3C82.2996 113.1 80.0996 109.9 80.0996 109.9Z"
                  stroke="black"
                  stroke-miterlimit="10"
                  stroke-width="1.5"
                />
                <path
                  d="M53.2988 70.6997C58.3988 73.4997 64.2988 77.8997 65.7988 79.9997C67.3988 82.2997 68.2988 84.3997 67.4988 87.0997C65.6988 92.9997 58.5988 89.1997 55.7988 86.4997C58.3988 88.7997 59.8988 92.1997 57.9988 95.3997"
                  fill="#FAC484"
                />
                <path
                  d="M53.2988 70.6997C58.3988 73.4997 64.2988 77.8997 65.7988 79.9997C67.3988 82.2997 68.2988 84.3997 67.4988 87.0997C65.6988 92.9997 58.5988 89.1997 55.7988 86.4997C58.3988 88.7997 59.8988 92.1997 57.9988 95.3997"
                  stroke="black"
                  stroke-miterlimit="10"
                  stroke-width="1.5"
                />
              </g>
              <defs>
                <clippath
                  id="clip0_1504_19506"
                >
                  <rect
                    fill="white"
                    height="79.5"
                    transform="translate(30 35)"
                    width="90.4"
                  />
                </clippath>
              </defs>
            </svg>
          </div>
          <div
            class="css-yzqqny"
          >
            <div
              class="css-0"
            >
              <div
                class="css-t01far"
              >
                <div
                  class="css-1wdjy8v"
                >
                  Tell Your Story (500 characters max)
                  <span
                    class="chakra-text css-1b8yx6g"
                  >
                     
                    *
                  </span>
                </div>
                <p
                  class="chakra-text css-dqpk00"
                >
                  Tell a compelling story of yourself to inspire potential fans.
                </p>
                <input
                  class="chakra-input css-12odza4"
                  name="story"
                  placeholder="Tell Your Story"
                  value=""
                />
              </div>
              <div
                class="css-lrby83"
              >
                <button
                  class="chakra-button css-k6pd2s"
                  type="button"
                >
                  SUBMIT
                  <span
                    class="css-9n3bbc"
                  />
                </button>
              </div>
            </div>
          </div>
          <button
            class="chakra-button css-1mv50kv"
            type="button"
          >
            SUBMIT
            <span
              class="css-9n3bbc"
            />
          </button>
        </div>
      </div>
      <p
        class="chakra-text css-1n4wwj3"
      />
    </div>
  </div>
</div>
`;
